python/src/aiconfig/default_parsers/parameterized_model_parser.py

```diff
@@ -117,13 +117,13 @@ def resolve_prompt_template(
         """
         return resolve_prompt_string(prompt, params, ai_config, prompt_template)
 
-    def get_prompt_template(self, prompt: Prompt, aiConfig: "AIConfigRuntime") -> str:
+    def get_prompt_template(self, prompt: Prompt, aiConfig: "AIConfigRuntime") -> str | None:
         """
         An overrideable method that returns a template for a prompt.
         """
         if isinstance(prompt.input, str):
             return prompt.input
-        elif isinstance(prompt.input, PromptInput) and isinstance(prompt.input.data, str):
+        if isinstance(prompt.input, PromptInput) and isinstance(prompt.input.data, str):
             return prompt.input.data
-        else:
```

Comment:
I left this in intentionally here. Because if a model parser says it's parameterized and we fail to resolve its prompt template, then we *should* throw an exception.

Reply:
That's perfect thanks!